src/search/videoId.ts

```diff
@@ -0,0 +1,31 @@
+import client from "../api/client.ts";
+import {
+  CHANNEL_ID,
+  MAX_RESULTS,
+  PART,
+  FIELDS,
+  TYPE,
+  VIDEO_DURATION,
+} from "../utils/constants.ts";
+
+async function getVideosId(): Promise<string[]> {
+  try {
+    const response = await client.search.list({
+      part: PART,
+      channelId: CHANNEL_ID,
+      fields: FIELDS,
+      maxResults: MAX_RESULTS,
+      type: TYPE,
+      videoDuration: VIDEO_DURATION,
+    });
+    return response.data.items
+      ? (response.data.items
+          .map((item) => item.id?.videoId)
+          .filter((videoId) => typeof videoId === "string") as string[])
+      : [];
+  } catch (err: any) {
+    throw new Error("Error fetching videos from the API: " + err.message);
+  }
+}
```

Comment:
Now that we are working on typescript let's talk about a better way to handle errors and error cases, please read about sum types or ADTs and let's try to apply them here to handle the errors